src/link/Wasm.zig

```diff
@@ -3363,6 +3367,36 @@ fn emitProducerSection(binary_bytes: *std.ArrayList(u8)) !void {
     );
 }
 
+fn emitBuildIdSection(binary_bytes: *std.ArrayList(u8)) !void {
+    const header_offset = try reserveCustomSectionHeader(binary_bytes);
+
+    const writer = binary_bytes.writer();
+    const build_id = "build_id";
+    try leb.writeULEB128(writer, @intCast(u32, build_id.len));
+    try writer.writeAll(build_id);
+
+    var id: [16]u8 = undefined;
+    std.crypto.hash.Blake3.hash(binary_bytes.items, &id, .{});
+    var uuid: [36]u8 = undefined; // UUIDv5 length is always 36
+    _ = try std.fmt.bufPrint(&uuid, "{s}-{s}-{s}-{s}-{s}", .{
```

Comment:
This is fine, but I also wonder if this could be more elegantly expressed as:
```suggestion
    const uuid = 
        hex(id[0..4]) ++ "-" ++
        hex(id[4..6]) ++ "-" ++
        hex(id[6..8]) ++ "-" ++
        hex(id[8..10]) ++ "-" ++
        hex(id[10..]);
```

Reply:
I'm a little bit confused here. Do we have such a `hex()` function?

Reply:
I'm sorry, I didn't mean my suggestion literally; merely as an illustrative example. I don't think we have a function like that in the std lib yet. Maybe something like this: 

```zig
pub fn hex(bytes: anytype) [bytes.len * 2]u8 {
    const hex_charset = "0123456789abcdef";
    var result: [bytes.len * 2]u8 = undefined;
    for (bytes, 0..) |b, i| {
        result[i * 2 + 0] = hex_charset[b >> 4];
        result[i * 2 + 1] = hex_charset[b & 15];
    }
    return result;
}
```